axonserver/src/main/java/io/axoniq/axonserver/rest/EventsRestController.java

```diff
@@ -238,44 +239,16 @@ private void completeEmitter(SseEmitter emitter) {
             @ApiIgnore final Authentication principal) {
-        auditLog.info("[{}@{}] Request to submit events.", AuditLog.username(principal), context);
-
         if (jsonEvents.messages.isEmpty()) {
             throw new IllegalArgumentException("Missing messages");
         }
-        CompletableFuture<Void> result = new CompletableFuture<>();
-        StreamObserver<InputStream> eventInputStream = eventStoreClient.appendEvent(context,
-                                                                                    getOrDefault(principal,
-                                                                                                 GrpcContextAuthenticationProvider.DEFAULT_PRINCIPAL),
-                                                                                    new StreamObserver<Confirmation>() {
-                                                                                        @Override
-                                                                                        public void onNext(
-                                                                                                Confirmation confirmation) {
-                                                                                            result.complete(null);
-                                                                                        }
-
-                                                                                        @Override
-                                                                                        public void onError(
-                                                                                                Throwable throwable) {
-                                                                                            result.completeExceptionally(
-                                                                                                    throwable);
-                                                                                        }
-
-                                                                                        @Override
-                                                                                        public void onCompleted() {
-                                                                                            // no action needed
-                                                                                        }
-                                                                                    });
-        if (eventInputStream != null) {
-            jsonEvents.messages.forEach(jsonEvent -> eventInputStream
-                    .onNext(new ByteArrayInputStream(jsonEvent.asEvent().toByteArray())));
-            eventInputStream.onCompleted();
-        }
-        return result;
+
+        Flux<SerializedEvent> events = Flux.fromStream(jsonEvents.messages.stream().map(jsonEvent -> new SerializedEvent(jsonEvent.asEvent())));
```

Comment:
It'd be nicer to put each method call in a new line. It will help us during stack trace analysis.